echemdb/data/cv/entry.py

```diff
@@ -100,6 +100,35 @@ def __getitem__(self, name):
         """
         return Descriptor(self.package.descriptor)[name]
 
+    def citation(self):
+        r"""
+        Returns a formated reference from the entrie's bibliography of style:
+    
+        J. Doe, et al., Journal Name, volume (YEAR) page, "Title"
+
+        EXAMPLES::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.citation()
+            "O. Alves *et al.*, **13** (2011) 6010–6021 '*Electrochemistry at Ru(0001) in a flowing CO-saturated electrolyte—reactive and inert adlayer phases*'"
+
+        """
+        from pybtex.textutils import abbreviate
+
+        def latex_to_text(latex_text):
+            from pylatexenc.latex2text import LatexNodes2Text
+            return LatexNodes2Text().latex_to_text(latex_text)
+
+        firstname_inititial = latex_to_text(abbreviate(self.bibliography.persons['author'][0].first_names[0]))
+        lastname = latex_to_text(self.bibliography.persons['author'][0].last_names[0])
+        etal = ' *et al.*' if len(self.bibliography.persons['author']) > 1 else ''
+        volume = latex_to_text(self.bibliography.fields['volume'])
+        year = latex_to_text(self.bibliography.fields['year'])
+        pages = latex_to_text(self.bibliography.fields['pages'])
+        title = latex_to_text(self.bibliography.fields['title'])
+        citation = f"{firstname_inititial} {lastname}{etal}, **{volume}** ({year}) {pages} '*{title}*'"
```

Comment:
Do we really want the title in single quotes? Maybe that's customary in your field, so just wondering.

Anyway, bad things will happen here if any of these strings (in particular) title contains anything that has a meaning in markdown, in particular of course `<`.

Should we try to use some library to construct such markdown snippets and hope that such library gets the escaping right?

Reply:
Actually, every journal has its own style. To be honest I don't have any preferences. Writing it only in italics would also be fine for me.

Reply:
Maybe this could solve the issue? https://pypi.org/project/markdown-strings/

Reply:
That package seems to store everything as plain strings, see https://github.com/awesmubarak/markdown_strings/issues/8. It might be good enough for our application but you cannot chain commands with an approach like this.
(I struggled with this a bit at https://github.com/standardese/standardese where I had to store everything as strings as well and eventually settled with parsing all arguments as MarkDown before processing them…)
Also, that project has not seen a release since two years almost.

Reply:
I wonder if we should put all MarkDown generation into proper templates, i.e., have a `citation.md` template that reads something like:
```md
{{ persons['author'].0.first_names.0 | abbreviate }} {{ persons['author'].0.last_names.0 }} {{ TODO: et al here }}, **{{ fields.volume }}** ({{ fields.year }}) {{ fields.pages }} "*{{ fields.title }}*"
```
That template could then be the string rendering of `.bibliography` which is currently not terribly useful.

Together with #78 that could work. This approach also solves #70 and #42.

Reply:
Maybe we should discuss this. Currently, I can't imagine how this works, i.e,,
* will `entry.citation` actually call `citation.md` if it is used not to generate the website but for example in a jupyter project?
* what is the difference in creating for example the title string in `entry.citation` than generating it directly in the markdown template as `{{ fields.title }}`. (should be `entry.fields.title`?)

Reply:
> will entry.citation actually call citation.md if it is used not to generate the website but for example in a jupyter project?

Not exactly. I guess that `str(entry.bibliograpy)` would render `citation.md` internally. There is no need for `entry.citation` anymore.

> what is the difference in creating for example the title string in entry.citation than generating it directly in the markdown template as {{ fields.title }}. (should be entry.fields.title?)

I am not entirely sure what is your question here. This is very similar, in one case, Python code is running in a `.py` file, in the other case very similar code is running in a template.

Historically, it turned out to be a good idea to use templates to create websites and not to produce strings that encode a website in code. Mostly, because templates are much easier to work with in practice.

The entire `cv_entry.md` could be  a method of `Entry` instead of being a template. In that case, it seems obvious (to me at least) that a template is the correct solution here. In the case of the citation I am less convinced that this has to be a template but since the code you wrote performs a lot of string manipulations, it feels reasonable to use a templating engine instead.

Reply:
Sorry, I forgot that there is no actual bibliography in the YAML.